### PR TITLE
Minor MPI cleanups

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

File: dc/core/MPIChannel.cpp [hunk 1]
```diff
@@ -177,6 +178,13 @@ void MPIChannel::receive( char* dataBuffer, const size_t messageSize,
     MPI_Status status;
     MPI_CHECK( MPI_Recv_Nospin( (void*)dataBuffer, messageSize, MPI_BYTE, src,
                                 tag, _mpiComm, &status ));
+
+    // Validate the numer of bytes received
```
Comment (tribal-tec) [inline dc/core/MPIChannel.cpp]: typo 'numer'

File: dc/core/MPINospin.cpp [hunk 2]
```diff
@@ -57,36 +57,36 @@ int MPI_Probe_Nospin( const int source, const int tag, MPI_Comm comm,
                       MPI_Status* status )
 {
     int flag = 0;
-    MPI_Iprobe( source, tag, comm, &flag, status );
+    int ret = MPI_Iprobe( source, tag, comm, &flag, status );
```
Comment (tribal-tec) [inline dc/core/MPINospin.cpp]: check ret before continuing/overwriting it in line 67?

**Comment by tribal-tec** [comment]
+1 otherwise

**Comment by rdumusc** [comment]
All done, thanks!

**Comment by rdumusc** [comment]
For consistency with the MVAPICH setup I also disbled the CPU affinity in openmpi + infiniband. This is complementary to the --bind-to none as also explained here (http://www.arc.ox.ac.uk/content/how-run-mpi-applications). This last change has no impact on our current setup.

**Pull request merged**